hare/algorithm_test.go

```diff
@@ -182,48 +178,59 @@ func (mev *mockEligibilityValidator) ValidateEligibilityGossip(context.Context,
+	c := config.Config{N: 10, F: 5, RoundDuration: time.Second, ExpectedLeaders: 5, LimitIterations: 1, LimitConcurrent: 1, Hdist: 20}
 	p := generateConsensusProcessWithConfig(t, c, make(chan any, 10))
 	p.Start()
-	time.Sleep(time.Duration(6*p.cfg.RoundDuration) * time.Second)
+	time.Sleep(6 * p.cfg.RoundDuration)
 
-	assert.EqualValues(t, 1, p.getRound()/4)
+	require.EqualValues(t, 1, p.getRound()/4)
 }
 
 func TestConsensusProcess_eventLoop(t *testing.T) {
-	ctrl := gomock.NewController(t)
-
 	net := &mockP2p{}
-	broker := buildBroker(t, t.Name())
-	broker.mockSyncS.EXPECT().IsSynced(gomock.Any()).Return(true).AnyTimes()
-	broker.mockSyncS.EXPECT().IsBeaconSynced(gomock.Any()).Return(true).AnyTimes()
-	broker.Start(context.Background())
-	t.Cleanup(broker.Close)
-	c := config.Config{N: 10, F: 5, RoundDuration: 2, ExpectedLeaders: 5, LimitIterations: 1000, LimitConcurrent: 1000, Hdist: 20}
-	c.F = 2
+	c := config.Config{N: 10, F: 5, RoundDuration: 2 * time.Second, ExpectedLeaders: 5, LimitIterations: 1000, LimitConcurrent: 1000, Hdist: 20}
 	proc := generateConsensusProcessWithConfig(t, c, make(chan any, 10))
 	proc.publisher = net
 
-	mo := mocks.NewMockRolacle(ctrl)
+	mo := mocks.NewMockRolacle(gomock.NewController(t))
 	mo.EXPECT().IsIdentityActiveOnConsensusView(gomock.Any(), gomock.Any(), proc.layer).Return(true, nil).Times(1)
 	mo.EXPECT().Proof(gomock.Any(), proc.layer, proc.getRound()).Return(nil, nil).Times(2)
 	mo.EXPECT().CalcEligibility(gomock.Any(), proc.layer, proc.getRound(), gomock.Any(), proc.nid, gomock.Any()).Return(uint16(1), nil).Times(1)
 	proc.oracle = mo
-
 	proc.value = NewSetFromValues(types.ProposalID{1}, types.ProposalID{2})
-	proc.cfg.F = 2
+
 	var wg sync.WaitGroup
 	wg.Add(1)
 	go func() {
 		proc.eventLoop()
 		wg.Done()
 	}()
 	time.Sleep(500 * time.Millisecond)
-	assert.Equal(t, 1, net.getCount())
+	require.Equal(t, 1, net.getCount())
 	proc.terminate()
 	wg.Wait()
 }
 
+// test that proc.Stop() actually returns and cause the consensus process to be GC'ed.
+func TestConsensusProcess_StartAndStop(t *testing.T) {
+	c := config.Config{N: 10, F: 5, RoundDuration: 50 * time.Millisecond, ExpectedLeaders: 5, LimitIterations: 1, LimitConcurrent: 1000, Hdist: 20}
+	proc := generateConsensusProcessWithConfig(t, c, make(chan any, 10))
+	proc.publisher = &mockP2p{}
+
+	mo := mocks.NewMockRolacle(gomock.NewController(t))
+	mo.EXPECT().IsIdentityActiveOnConsensusView(gomock.Any(), gomock.Any(), proc.layer).Return(true, nil).AnyTimes()
+	mo.EXPECT().Proof(gomock.Any(), gomock.Any(), gomock.Any()).Return(nil, nil).AnyTimes()
+	mo.EXPECT().CalcEligibility(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), proc.nid, gomock.Any()).Return(uint16(1), nil).AnyTimes()
+	proc.oracle = mo
+
+	proc.value = NewSetFromValues(types.ProposalID{1}, types.ProposalID{2})
+	proc.Start()
+	require.Eventually(t, func() bool {
+		proc.Stop()
+		return true
+	}, 500*time.Millisecond, 100*time.Millisecond)
```

Comment:
It could be simplified to:
```suggestion
	proc.Stop()
```

The test will timeout if `proc.Stop()` never returns.

Reply:
i don't see why test timing out is better. when i do `go test` it will just hang.